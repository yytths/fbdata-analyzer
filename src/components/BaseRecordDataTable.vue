<template>
  <v-card>
    <v-card-title>
      <v-spacer></v-spacer>
      <v-text-field
        v-model="search"
        append-icon="mdi-magnify"
        label="Search"
        single-line
        hide-details
      ></v-text-field>
    </v-card-title>

    <v-data-table
      :search="search"
      :headers="headers"
      :items="items"
      :items-per-page="5"
    >
      <template v-slot:no-data>
        レコードがありません
      </template>
      <template v-slot:no-results>
        レコードがみつかりません
      </template>

      <!-- 親コンポーネントからVuetifyのdatatableのslotを操作する -->
      <template v-for="(slotContent, slotName) of $scopedSlots" #[slotName]="data">
        <slot :name="slotName" v-bind="data"/>
      </template>
    </v-data-table>
  </v-card>
</template>

<script>
export default {
  name: 'BaseRecordDataTable',

  props: {
    items: {
      default() { return []; },
      type: Array,
    },
    headers: {
      default() { return []; },
      type: Array,
    },
  },

  data: () => ({
    search: '',
  }),
};
</script>
