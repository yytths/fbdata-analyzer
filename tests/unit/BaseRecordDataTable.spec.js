import Vue from 'vue';
import Vuetify from 'vuetify';
import {
  mount,
  createLocalVue,
} from '@vue/test-utils';
import BaseRecordDataTable from '@/components/BaseRecordDataTable.vue';

Vue.use(Vuetify);
const localVue = createLocalVue(Vue);

describe('BaseRecordDataTable.vue', () => {
  let vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  it('normal test. 1 records.', () => {
    const headers = [
      {
        text: 'テキスト1',
        value: 'textOne',
      },
      {
        text: 'テキスト2',
        value: 'textTwo',
      },
    ];
    const items = [
      {
        textOne: 1,
        textTwo: 2,
      }
    ];
    const wrapper = mount(BaseRecordDataTable, {
      localVue,
      vuetify,
      propsData: {
        headers,
        items,
      }
    });
    // With jest we can create snapshot files of the HTML output
    expect(wrapper.html()).toMatchSnapshot();

    const firstColumnHeaderElement = wrapper.find('.v-data-table-header > tr > th > span');
    expect(firstColumnHeaderElement.text()).toBe('テキスト1');

    const firstColumnValueElement = wrapper.find('.v-data-table__wrapper > table > tbody > tr > td');
    expect(firstColumnValueElement.text()).toBe('1');
  });

  it('normal test. no record message.', () => {
    const expected = 'レコードがありません';

    const wrapper = mount(BaseRecordDataTable, {
      localVue,
      vuetify,
    });
    // With jest we can create snapshot files of the HTML output
    expect(wrapper.html()).toMatchSnapshot();

    const noDataText = wrapper.find('.v-data-table__empty-wrapper > td');
    expect(noDataText.text()).toBe(expected);

    // TODO: WARNをなくすために、@vue/test-utils@1.0.0に更新する。
    // https://github.com/vuejs/vue-test-utils/issues/1407
  });
});
