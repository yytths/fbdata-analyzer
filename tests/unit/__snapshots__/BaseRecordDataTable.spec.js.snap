// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseRecordDataTable.vue normal test. 1 records. 1`] = `
<div class="v-card v-sheet theme--light">
  <div class="v-card__title">
    <div class="spacer"></div>
    <div class="v-input v-input--hide-details theme--light v-text-field v-text-field--single-line">
      <div class="v-input__control">
        <div class="v-input__slot">
          <div class="v-text-field__slot"><label for="input-4" class="v-label theme--light" style="left: 0px; position: absolute;">Search</label><input id="input-4" type="text"></div>
          <div class="v-input__append-inner">
            <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-magnify theme--light"></i></div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="v-data-table theme--light">
    <div class="v-data-table__wrapper">
      <table>
        <colgroup>
          <col class="">
          <col class="">
        </colgroup>
        <thead class="v-data-table-header">
          <tr>
            <th role="columnheader" scope="col" aria-label="テキスト1: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>テキスト1</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
            <th role="columnheader" scope="col" aria-label="テキスト2: Not sorted. Activate to sort ascending." aria-sort="none" class="text-start sortable"><span>テキスト2</span><i aria-hidden="true" class="v-icon notranslate v-data-table-header__icon mdi mdi-arrow-up theme--light" style="font-size: 18px;"></i></th>
          </tr>
        </thead>
        <tbody>
          <tr class="">
            <td class="text-start">1</td>
            <td class="text-start">2</td>
          </tr>
        </tbody>
      </table>
    </div>
    <div class="v-data-footer">
      <div class="v-data-footer__select">Rows per page:<div class="v-input v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light v-text-field v-select">
          <div class="v-input__control">
            <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-13" class="v-input__slot">
              <div class="v-select__slot">
                <div class="v-select__selections">
                  <div class="v-select__selection v-select__selection--comma">5</div><input aria-label="$vuetify.dataTable.itemsPerPageText" id="input-13" readonly="readonly" type="text" aria-readonly="false" autocomplete="off">
                </div>
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                </div><input type="hidden" value="5">
              </div>
              <div class="v-menu">
                <!---->
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="v-data-footer__pagination">1-1 of 1</div>
      <div class="v-data-footer__icons-before"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="Previous page"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-left theme--light"></i></span></button></div>
      <div class="v-data-footer__icons-after"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="Next page"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-right theme--light"></i></span></button></div>
    </div>
  </div>
</div>
`;

exports[`BaseRecordDataTable.vue normal test. no record message. 1`] = `
<div class="v-card v-sheet theme--light">
  <div class="v-card__title">
    <div class="spacer"></div>
    <div class="v-input v-input--hide-details theme--light v-text-field v-text-field--single-line">
      <div class="v-input__control">
        <div class="v-input__slot">
          <div class="v-text-field__slot"><label for="input-24" class="v-label theme--light" style="left: 0px; position: absolute;">Search</label><input id="input-24" type="text"></div>
          <div class="v-input__append-inner">
            <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-magnify theme--light"></i></div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div class="v-data-table theme--light">
    <div class="v-data-table__wrapper">
      <table>
        <colgroup></colgroup>
        <thead class="v-data-table-header">
          <tr></tr>
        </thead>
        <tbody>
          <tr class="v-data-table__empty-wrapper">
            <td colspan="0">
              レコードがありません
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div class="v-data-footer">
      <div class="v-data-footer__select">Rows per page:<div class="v-input v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light v-text-field v-select">
          <div class="v-input__control">
            <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-31" class="v-input__slot">
              <div class="v-select__slot">
                <div class="v-select__selections">
                  <div class="v-select__selection v-select__selection--comma">5</div><input aria-label="$vuetify.dataTable.itemsPerPageText" id="input-31" readonly="readonly" type="text" aria-readonly="false" autocomplete="off">
                </div>
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
                </div><input type="hidden" value="5">
              </div>
              <div class="v-menu">
                <!---->
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="v-data-footer__pagination">–</div>
      <div class="v-data-footer__icons-before"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="Previous page"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-left theme--light"></i></span></button></div>
      <div class="v-data-footer__icons-after"><button type="button" disabled="disabled" class="v-btn v-btn--disabled v-btn--flat v-btn--icon v-btn--round v-btn--text theme--light v-size--default" aria-label="Next page"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate mdi mdi-chevron-right theme--light"></i></span></button></div>
    </div>
  </div>
</div>
`;
